tests/components/modbus/test_modbus.py

```diff
@@ -17,56 +21,265 @@
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                }
+            ]
+        },
+        {
+            DOMAIN: [
+                {
+                    CONF_TYPE: "udp",
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                    CONF_NAME: "modbusTest",
+                    CONF_TIMEOUT: 30,
+                    CONF_DELAY: 10,
+                }
+            ]
+        },
+        {
+            DOMAIN: [
+                {
+                    CONF_TYPE: "rtuovertcp",
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                }
+            ]
+        },
+        {
+            DOMAIN: [
+                {
+                    CONF_TYPE: "rtuovertcp",
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                    CONF_NAME: "modbusTest",
+                    CONF_TIMEOUT: 30,
+                    CONF_DELAY: 10,
+                }
+            ]
+        },
+        {
+            DOMAIN: [
+                {
+                    CONF_TYPE: "serial",
+                    CONF_BAUDRATE: 9600,
+                    CONF_BYTESIZE: 8,
+                    CONF_METHOD: "rtu",
+                    CONF_PORT: "usb01",
+                    CONF_PARITY: "E",
+                    CONF_STOPBITS: 1,
+                }
+            ]
+        },
+        {
+            DOMAIN: [
+                {
+                    CONF_TYPE: "serial",
+                    CONF_BAUDRATE: 9600,
+                    CONF_BYTESIZE: 8,
+                    CONF_METHOD: "rtu",
+                    CONF_PORT: "usb01",
+                    CONF_PARITY: "E",
+                    CONF_STOPBITS: 1,
+                    CONF_NAME: "modbusTest",
+                    CONF_TIMEOUT: 30,
+                    CONF_DELAY: 10,
+                }
+            ]
         },
     ],
 )
+async def test_config_modbus(hass, do_config):
+    """Run test for modbus."""
+
+    mock_sync = mock.MagicMock()
+    with mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusTcpClient", return_value=mock_sync
+    ), mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusSerialClient",
+        return_value=mock_sync,
+    ), mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusUdpClient", return_value=mock_sync
+    ):
+        # mocking is needed to secure the pymodbus library is not called!
+        assert await async_setup_component(hass, DOMAIN, do_config)
+        await hass.async_block_till_done()
+        assert DOMAIN in hass.data
+
+        # the "if" should really be in the do_config above, but
+        # there are no easy way to have vol add default in a test setup
+        if CONF_NAME not in do_config[DOMAIN][0]:
+            do_config[DOMAIN][0][CONF_NAME] = DEFAULT_HUB
+        assert do_config[DOMAIN][0][CONF_NAME] in hass.data[DOMAIN]
```

Comment:
We want to avoid asserting integration details like hass.data. What do we want to check with this test? Usually we assert that the library (patched) is called as expected.

Reply:
How can I check if the integration was successful ? I thought the only way was to examine hass.data, but if you can give me another way I will change the code.

I am trying load multiple instances of modbus, and want to check that it works.

Reply:
We can assert the state of the config entry instance (`entry.state`) of we want to check that the config entry is loaded (`ENTRY_STATE_LOADED`).

Besides that we can check the calls to the library as mentioned.